Dropdown/src/Dropdown.jsx

```diff
@@ -171,30 +171,36 @@ class Dropdown extends Component {
     }
 
     searchItems() {
+
+        const { props } = this;
+
         let closestValueLength = 0, closestValue = null, itemIndex = 0;
 
-        this.props.options.forEach((option, index) => {
+        props.options.forEach((option, index) => {
             let regex = this.state.dropdownText.replace(/([.?*+^$[\]\\(){}|-])/g, "\\$1");
 
+            const label = props.decode ? props.decode(option.label) : option.label;
+
             let stringToMatchBeginning = new RegExp("^" + regex, "gi");
 
-            let labelToMatch = typeof option.label === "string" ? option.label : (option.searchableValue || "");
+            let labelToMatch = typeof label === "string" ? label : (option.searchableValue || "");
 
             if (labelToMatch.match(stringToMatchBeginning) && labelToMatch.match(stringToMatchBeginning).length > closestValueLength) {
                 closestValueLength = labelToMatch.match(stringToMatchBeginning).length;
                 closestValue = option.value;
                 itemIndex = index;
-
             }
         });
 
         if (closestValue === null) {
-            this.props.options.forEach((option, index) => {
+            props.options.forEach((option, index) => {
```

Comment:
Does this `forEach` look very similar to the one above? Maybe they could be extracted to a function?